server/server.go

```diff
@@ -124,42 +126,160 @@ type Server struct {
 	BuildInfo         chronograf.BuildInfo
 }
 
-func provide(p oauth2.Provider, m oauth2.Mux, ok func() bool) func(func(oauth2.Provider, oauth2.Mux)) {
+func provide(p oauth2.Provider, m oauth2.Mux, ok func() error) func(func(oauth2.Provider, oauth2.Mux)) {
 	return func(configure func(oauth2.Provider, oauth2.Mux)) {
-		if ok() {
+		if err := ok(); err == nil {
 			configure(p, m)
 		}
 	}
 }
 
 // UseGithub validates the CLI parameters to enable github oauth support
-func (s *Server) UseGithub() bool {
-	return s.TokenSecret != "" && s.GithubClientID != "" && s.GithubClientSecret != ""
+func (s *Server) UseGithub() error {
+	errMsg := []string{}
+
+	switch {
```

Comment:
stylistic nit (may not be applicable to golang):

`switch` statements (to me at least) usually imply a `default` case, but the default cause here is to execute the function, which makes these more like [guard clauses,](https://wiki.c2.com/?GuardClause) and it seems like this would be better as two `if` statements, since they both early return anyway.